### PR TITLE
Crusader

### PR DESCRIPTION
**Pull request by Radiating8**
whoops, wrong branch

### CODE REVIEW

File: src/lib/mafia/roles/town/Crusader.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		}
+
+		// Select visitor to be killed.
+		let visitors: Set<Player> = new Set();
+		let actionNames = [...playerRecord.keys()];
+		for (let i = 0; i < playerRecord.size; i++) {
+			let actionName = actionNames[i];
+			playerRecord.get(actionName).by.forEach(visitor => visitors.add(visitor));
+		}
+		let playerToKill: Player = Array.from(visitors)[Math.floor(Math.random() * visitors.size)];
```
Comment (Soumil07) [inline src/lib/mafia/roles/town/Crusader.ts:29]: Use `util.randomArray` here: https://github.com/Soumil07/godfather/blob/master/src/lib/util/utils.ts#L28

File: src/lib/mafia/roles/town/Crusader.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+	public runAction(actions: NightActionsManager, target: Player) {
+		const playerRecord = actions.record.get(target.user.id);
+		if (playerRecord.size === 0) {
+			return;
+		}
+
+		// Select visitor to be killed.
+		let visitors: Set<Player> = new Set();
+		let actionNames = [...playerRecord.keys()];
```
Comment (Soumil07) [inline src/lib/mafia/roles/town/Crusader.ts:24]: alternatively, just `const { visitors } = this.player`. For this to work though, you'll have to put Crusader's priority dead last, next to Ambusher.

File: src/lib/mafia/roles/town/Crusader.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		}
+	}
+
+	public canTarget(player: Player) {
+		// TODO: customizable rule here
+		if (player === this.player) return { check: false, reason: 'You cannot target yourself.' };
+		return super.canTarget(player);
+	}
+
+	public actionConfirmation(target: Player) {
```
Comment (Soumil07) [inline src/lib/mafia/roles/town/Crusader.ts:65]: This can be safely removed.

**Review by Soumil07 — CHANGES_REQUESTED** [review]

**Review by MeDerpWasTaken — CHANGES_REQUESTED** [review]
Don't use killed, use something like attacked instead

**Review by MeDerpWasTaken — APPROVED** [review]

**Review by Soumil07 — APPROVED** [review]

**Pull request merged**